### PR TITLE
Issue #7077: correct docs for UnnecessarySemicolonAfterTypeMemberDeclaration

### PR DESCRIPTION
**Pull request by strkkk**
#7077

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java [hunk 1]
```diff
@@ -29,6 +29,8 @@
 /**
  * <p>
  * Checks if unnecessary semicolon is used after type member declaration.
+ * Note: this check is not applicable to empty statements (unnecessary semicolons inside methods or
+ * init blocks), {@code EmptyStatementCheck} is responsible for it.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java]: Please make a link to the check.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java [hunk 2]
```diff
@@ -29,6 +29,8 @@
 /**
  * <p>
  * Checks if unnecessary semicolon is used after type member declaration.
+ * Note: this check is not applicable to empty statements (unnecessary semicolons inside methods or
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java]: @pbludov Should the note be in a note section?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java (hunk 1 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java]: Can you show an example how this link should look like?
If in javadoc it is `{@link}` how it should look in xml?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java (hunk 2 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java]: Forgot about it, changed

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java (hunk 1 above)
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java]: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/ClassMemberImpliedModifierCheck.java#L34
https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/config_modifier.xml#L33

Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java]: done

**Pull request merged**